modules/@apostrophecms/notification/index.js

```diff
@@ -156,22 +156,19 @@ module.exports = {
       // Resolves with an object with `notifications` and `dismissed`
       // properties.
       //
-      // If `options.displayingIds` is set, notifications
-      // whose `_id` properties appear in it are not returned.
+      // If `options.modifiedOnOrSince` is set, notifications
+      // greater than the timestamp are sent.
 
       async find(req, options) {
         try {
           const notifications = await self.db.find({
-            userId: req.user._id
+            userId: req.user._id,
+            ...(options.modifiedOnOrSince && { createdAt: { $gt: new Date(options.modifiedOnOrSince) } })
           }).sort({ createdAt: 1 }).toArray();
+
           return {
-            notifications: _.filter(notifications, function (notification) {
-              if (options.displayingIds && options.displayingIds.length > 0) {
-                return !_.includes(options.displayingIds || [], notification._id);
-              }
-              return notification;
-            }),
-            dismissed: _.difference(options.displayingIds || [], _.map(notifications, '_id'))
+            notifications,
+            dismissed: []
```

Comment:
Hmm, this is a problem actually, how will we know a notification currently on the screen has been dismissed, if that happens on another screen?

I think the answer is that when a notification is dismissed, we set a `dismissed` flag on it and modify `updatedAt` (like we should always do when updating a notification). But we also remove all `dismissed` notifications on an interval when `updatedAt` is old enough, let's say an hour.

Reply:
... Then the frontend code will receive it like any other updated notification, but will also see the dismissed flag.

Reply:
... So that means it *is* time to implement updatedAt now 😄